pkg/pmem-csi-driver/pmem-csi-driver.go

```diff
@@ -193,34 +192,39 @@ func (pmemd *pmemDriver) Run() error {
 	return nil
 }
 
-func (pmemd *pmemDriver) registerNodeController() error {
-	fmt.Printf("Connecting to Registry at : %s\n", pmemd.cfg.RegistryEndpoint)
+func (pmemd *pmemDriver) registerNodeController() {
 	var err error
 	var conn *grpc.ClientConn
 
+	req := registry.RegisterControllerRequest{
+		NodeId:   pmemd.cfg.NodeID,
+		Endpoint: pmemd.cfg.ControllerEndpoint,
+	}
+
 	for {
+		glog.Infof("Connecting to registry server at: %s\n", pmemd.cfg.RegistryEndpoint)
 		conn, err = pmemgrpc.Connect(pmemd.cfg.RegistryEndpoint, pmemd.clientTLSConfig, connectionTimeout)
 		if err == nil {
-			glog.Info("Connected to RegistryServer!!!")
 			break
 		}
 		/* TODO: Retry loop */
-		glog.Infof("Failed to connect RegistryServer : %s, retrying after 10 seconds...", err.Error())
-		time.Sleep(10 * time.Second)
+		glog.Infof("Failed to connect registry server: %s, retrying after %v seconds...", err.Error(), retryTimeout.Seconds())
```

Comment:
What is this TODO about now? Still valid or has this been solved?

Reply:
yes, we can remove. I will do that